### PR TITLE
Feature/190 make aptatrans sklearn compatible

### LINKED ISSUE #190: [ENH] Changes to `AptaTransPipeline` to make it sklearn-like
The current design of `AptaTransPipeline` appears inconsistent with the expected scikit-learn–style API since it lacks `fit` and `predict` methods. Without these common interfaces, it’s unclear how benchmarking is intended to operate, especially given that `AptaNetPipeline` does follow this pattern: there’s no shared method that can be called on both classes for fitting and generating predictions. Hence, we should add `fit` and `predict` methods to `AptaTransPipeline`.

Additionally:
1. The description of `prot_words` should be standardized and rewritten in a clear, generic way across all instances where it appears, as its current form is overly confusing.
2. The `predict` function should explicitly document the datatype returned within the Tensor; at present, users have to inspect the `evaluate` function to infer what is being returned.

### PR DESCRIPTION
**Pull request by NennoMP**
This PR resolves #190.

Added `fit(...)` and `predict(...)` methods to `AptaTransPipeline` to make it sklearn compatible. Here is a summary of the changes to facilitate review:

* Changes to the **pipeline** (`_pipeline.AptaTransPipeline`):
  * Added a `fit(...)` method with sklearn-like interface to enable training;
  * Added a `predict(...)` method with sklearn-like interface to enable prediction.
  
* Changes to **lightning training code** (`_model_lightning.AptaTransLightning`, `_model_lightning.AptaTransEncoderLightning`):
  * Added a `predict_step(...)` method to enable inference when ground-truth labels are not available (since sklearn predict interface does not expect them);
  * Modified `_step(...)` method accordingly to handle both cases when ground-truth labels are or are not provided.
  
* Changes to **dataloaders** (`_api.APIDataset`):
  * Similarly to above, changes were needed to handle the scenario where ground-truth labels are not provided/available.
  
Finally, tests were updated accordingly.

### CODE REVIEW

File: pyaptamer/aptatrans/_model_lightning.py [hunk 1]
```diff
@@ -76,31 +75,44 @@ def __init__(
         self.weight_decay = weight_decay
         self.betas = betas
 
-    def _step(self, batch: tuple[Tensor, Tensor, Tensor], batch_idx: int) -> Tensor:
-        """Defines a single (mini-batch) step in the training/test loop.
+    def _step(
+        self,
+        batch: tuple[Tensor, Tensor] | tuple[Tensor, Tensor, Tensor],
+        batch_idx: int,
```
Comment (siddharth7113) [inline pyaptamer/aptatrans/_model_lightning.py:81]: Why are we passing `batch_idx` , I couldn't find the use in this particular function?

File: pyaptamer/aptatrans/_model_lightning.py [hunk 2]
```diff
@@ -143,13 +155,35 @@ def test_step(self, batch: tuple[Tensor, Tensor, Tensor], batch_idx: int) -> Ten
...
         """
-        loss, accuracy = self._step(batch, batch_idx)
+        _, loss, accuracy = self._step(batch, batch_idx)
 
         self.log("test_loss", loss, on_epoch=True, on_step=False, prog_bar=True)
         self.log("test_accuracy", accuracy, on_epoch=True, on_step=False, prog_bar=True)
 
         return loss
 
+    def predict_step(
```
Comment (siddharth7113) [inline pyaptamer/aptatrans/_model_lightning.py:165]: Would this function act as a replacement of test_step,

instead of using `test_step` to calculate loss, you call `predict_step` use it's output and ground_truth and then calculate loss , optimizier, backward pass and so on?

If so, can you open an issue about the same?

File: pyaptamer/aptatrans/_model_lightning.py [hunk 3]
```diff
@@ -256,45 +291,58 @@ def __init__(
         self.weight_mlm = weight_mlm
         self.weight_ssp = weight_ssp
 
-    def _step(self, batch: tuple[Tensor, Tensor, Tensor], batch_idx: int) -> Tensor:
-        """Defines a single (mini-batch) step in the training/test loop.
+    def _step(
+        self,
+        batch: tuple[Tensor, Tensor] | tuple[Tensor, Tensor, Tensor, Tensor],
+        batch_idx: int,
```
Comment (siddharth7113) [inline pyaptamer/aptatrans/_model_lightning.py:297]: batch_idx is not being used here in the function

**Review by siddharth7113 — COMMENTED** [review]
Overall looks good for now, but at later step, instead of having a test step we can use predict and use a seaprate loss calculation and optimiser loop like pytorch? (suggestion)

apart from couple of comments about `batch_idx`, this could be merged for now.

**Comment by siddharth7113** [comment]
@NennoMP  since this PR has been pending from long time and the structure has changed since then, could we close this and add instead detailed description in issue , open for new contributors.

**Comment by NennoMP** [comment]
@siddharth7113 I think so. A lot of stuff changed so it may be better to start fresh I guess? I would still reference this since I think the changes to the pipeline are still relevant/good.

**Comment by siddharth7113** [comment]
> @siddharth7113 I think so. A lot of stuff changed so it may be better to start fresh I guess? I would still reference this since I think the changes to the pipeline are still relevant/good.

Agreed, can you just make a new issue , mention this PR there, along with all the references and details, and maybe add labels ,good-first-issue` and probably `medium` or `hard` , otherwise there would be a in-flux of PR on this.
